src/host/ledger.h

```diff
@@ -954,17 +970,30 @@ namespace asynchost
       return last_idx;
     }
 
-    void truncate(size_t idx)
+    void truncate(size_t idx, bool complete = false)
```

Comment:
`complete` could be a little confusing, I'd prefer `complete_chunk`, there's also a `complete_ledger` further down